host/xtest/regression_1000.c

```diff
@@ -1744,3 +1746,43 @@ static void xtest_tee_test_1022(ADBG_Case_t *c)
 }
 ADBG_CASE_DEFINE(regression, 1022, xtest_tee_test_1022,
 		"Test dlopen()/dlsym()/dlclose() API");
+
+static void xtest_tee_test_1023(ADBG_Case_t *c)
+{
+	TEEC_Session session = {};
+	uint32_t ret_orig = 0;
+
+	Do_ADBG_BeginSubCase(c, "TPM test create session");
+	{
+		if (!ADBG_EXPECT_TEEC_SUCCESS(c,
+		    xtest_teec_open_session(&session, &tpm_log_test_ta_uuid,
+					    NULL, &ret_orig)))
+			return;
+	}
+	Do_ADBG_EndSubCase(c, "TPM test create session");
+
+	Do_ADBG_BeginSubCase(c, "TPM test service invocation");
+	{
+		TEEC_Operation op = TEEC_OPERATION_INITIALIZER;
```

Comment:
If no operation is needed, you should pass NULL to `TEEC_InvokeCommand()`.

But, since we're driving everything from the client app (this code), wouldn't it be simpler to have the TA just forward the operation to the PTA? Basically, `TA_InvokeCommandEntryPoint()` below would do what `invoke_system_pta()` currently does.

Reply:
> But, since we're driving everything from the client app (this code), wouldn't it be simpler to have the TA just forward the operation to the PTA? Basically, TA_InvokeCommandEntryPoint() below would do what invoke_system_pta() currently does.

So you mean to move most of the logic to test the event log inside my implementation of TA_InvokeCommandEntryPoint() to the regression test and just forward the commands from the PA to the TPA, right?
I actually considered that option, but I rejected it as:

1. I wanted to keep the regression test as simple as possible as I noticed that most of the tests on regression_1000.c were relatively simple.

2. I decided that the logic to check that a TPM event log was properly read was best placed inside the TA. On a real fTPM implementation, the event log would be processed inside the TA and not forwarded to the client app, so I didn't see a reason why I should move that logic to the client and then forward the event log there.

Cheers,
Javier

Reply:
Correct, that's what I meant.

I have no strong opinion either way, so it is enough for me to know that you considered both options and chose the one you liked most :)

Reply:
My comment about removing `op` if you don't need it still holds however.